### ISSUE TITLE
What's the issue with newer SDCC? IS there an upstream bug report?

### ISSUE THREAD

**Issue by spth** [cpldcpu/BitNetPDK#1]
The README mentions "Use SDCC 4.1.0 to build. Later versions create broken code". What is the problem here? Is there a corresponding ticket in the SDCC bug tracker?

**Comment by cpldcpu** [comment]
It's the same issue that I brought up in the free-pdk examples. I haven't been able to pinpoint the error back then.

**Comment by spth** [comment]
The one that turned out to be a mov vs. mov.io issue: https://github.com/free-pdk/free-pdk-examples/issues/13 ?

**Comment by cpldcpu** [comment]
Yeah, the compatibility-breaking change seems to have been one regarding IO instructions. 

My old UART routines compiled without error, but resulted in nonfunctional code. Very annoying and difficult to debug. I solved it by reverting to an older version of SDCC. I may need to find some time to figure out how to make it compatible to the newer SDCC versions.

**Comment by spth** [comment]
Then the solution should be easy: in all the I/O that you do via inline asm, change the instructions used to mov.io, set0.io, etc.
My recommendation: except for the few places where inline asm really provides an advantage, just use C.

**Comment by cpldcpu** [comment]
Just unrelated topic on this side: Any plans to migrate SDCC to git or even github? Would simplify submission of error reports significantly. Nowadays, I (and many others) have very little cross section to SVN and sourceforge and having to work with an unfamiliar system adds significant friction.

**Comment by spth** [comment]
AFAIK, there are no plans for a migration. But it looks like we'll get an official git mirror of the svn repo.

**Comment by spth** [comment]
Please check if the free-pdk examples work for you now. If yes, you can port the fixes there to here.
